### PR TITLE
Fix logout issue for organization switch tokens when authenticated user id set to null

### PR DESCRIPTION
**Pull request by sadilchamishka**

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java [hunk 1]
```diff
@@ -323,7 +323,11 @@ private void revokeTokensOfBindingRef(AuthenticatedUser user, String tokenBindin
                                     user.getFederatedIdPName(), authenticatedUser.getFederatedIdPName())
                             && StringUtils.equalsIgnoreCase(user.getUserName(), authenticatedUser.getUserName())) {
                         revokeFederatedTokens(consumerKey, user, accessTokenDO, tokenBindingReference);
-                    } else if (StringUtils.equalsIgnoreCase(userId, authenticatedUser.getUserId())) {
+                    } else if (StringUtils.equalsIgnoreCase(userId, authenticatedUser.getAuthenticatedSubjectIdentifier().split("@")[0])) {
+                        accessTokenDO.getAuthzUser().setUserId(userId);
+                        revokeTokens(consumerKey, accessTokenDO, tokenBindingReference);
```
Comment (inthirakumaaran) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java:328]:
```suggestion
                        revokeFederatedTokens(consumerKey, user, accessTokenDO, tokenBindingReference);
```

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java [hunk 2]
```diff
@@ -323,7 +323,11 @@ private void revokeTokensOfBindingRef(AuthenticatedUser user, String tokenBindin
                                     user.getFederatedIdPName(), authenticatedUser.getFederatedIdPName())
                             && StringUtils.equalsIgnoreCase(user.getUserName(), authenticatedUser.getUserName())) {
                         revokeFederatedTokens(consumerKey, user, accessTokenDO, tokenBindingReference);
-                    } else if (StringUtils.equalsIgnoreCase(userId, authenticatedUser.getUserId())) {
+                    } else if (StringUtils.equalsIgnoreCase(userId, authenticatedUser.getAuthenticatedSubjectIdentifier().split("@")[0])) {
```
Comment (inthirakumaaran) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java:326]: Shall we add another condition to limit this logic for B2B scenario, as in the ideal federated user case userID should be there

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /*
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
```
Comment (inthirakumaaran) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java:2]:
```suggestion
 * Copyright (c) 2020-2023, WSO2 LLC. (http://www.wso2.com).
```

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java (hunk 2 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java]: From the authenticated user, there is no such attribute to check whether a B2B scenario. In these else blocks, if the userId is same as the authenticated users userId, the token revocation should occur. But here userId is set to null. In order to complete the flow without any error, this workaround was suggested. Here if the authenticated subject identifer has a hint for the userId, then let the token to be revoked.

**Comment by sadilchamishka** [comment]
Closing due to this PR

**Pull request closed without merging**